model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -380,6 +381,35 @@ object DockstoreMethod {
 
 }
 
+case class DockstoreToolsMethod(methodPath: String, methodVersion: String) extends MethodRepoMethod {
+  override def validate: Option[DockstoreToolsMethod] = {
+    if (methodPath.nonEmpty && methodVersion.nonEmpty) Some(this) else None
+  }
+
+  override def methodUri: String = {
+    if (validate.isDefined) {
+      s"${repo.scheme}://${URLEncoder.encode(methodPath, UTF_8.name)}/${URLEncoder.encode(methodVersion, UTF_8.name)}"
+    } else {
+      throw new RawlsException(s"Could not generate a method URI from DockstoreToolsMethod with path \'$methodPath\', version \'$methodVersion\'")
+    }
+  }
+
+  override def repo: MethodRepository = DockstoreTools
+}
+
+object DockstoreToolsMethod {
+  def apply(uri: String): DockstoreToolsMethod = {
```

Comment:
Can we DRY out the `apply` method between `DockstoreToolsMethod` and `DockstoreMethod`? Looks similar or identical.

Reply:
Fair enough, I'll pull out a helper function.